tasks/vagrant.py

```diff
@@ -31,23 +31,54 @@ def __setup_provision(task):
     This tries to execute the provision twice due to
     problems described in issue #20
     """
+    if task.action_name == 'ad':
+        vagrant_up_retries = 2
+        vagrant_provision_retries = 3
+        provision_delay = 120
+    else:
+        vagrant_up_retries = 0
+        vagrant_provision_retries = 1
+        provision_delay = 0
+    retry_delay = 10
+
     task.execute_subtask(
         VagrantBoxDownload(
             box_name=task.template_name,
             box_version=task.template_version,
             link_image=task.link_image,
             timeout=None))
-    try:
-        task.execute_subtask(
-            VagrantUp(action_name=task.action_name, timeout=None))
-        task.execute_subtask(VagrantProvision(timeout=None))
-    except Exception as exc:
-        logging.debug(exc, exc_info=True)
-        logging.info("Failed to provision/up VM. Trying it again")
-        task.execute_subtask(VagrantCleanup(raise_on_err=False))
-        task.execute_subtask(
-            VagrantUp(action_name=task.action_name, timeout=None))
-        task.execute_subtask(VagrantProvision(timeout=None))
+
+    while True:
+        try:
+            task.execute_subtask(
+                VagrantUp(timeout=None))
+            break
+        except PopenException:
+            if vagrant_up_retries:
+                logging.info(
+                    "Retrying to bring the machine up, %s retries left" %
```

Comment:
nitpick: Could you please pass the variables as arguments to the logging functions or use the `str.format()`.
```python
logging.info("Retrying to bring the machine up, %s retries left", vagrant_up_retries)
```